packages/parser/lib/src/parser/grammar.dart

```diff
@@ -659,9 +659,9 @@ class ParserGrammar {
             _id++,
             ifKeyword: left[0] as IfKeywordToken,
             condition: value,
-            thenExpression: right[1] as Expression,
+            thenBody: right[1] as LambdaLiteral,
             elseKeyword: elsePart?.elementAt(1) as ElseKeywordToken,
-            elseExpression: elsePart?.elementAt(3) as Expression,
+            elseBody: elsePart?.elementAt(3) as LambdaLiteral,
```

Comment:
Do we want to constrain those to lambda-blocks? Earlier, we also allowed something like `if (true) a else b`

Reply:
We'll leave this as it is for now and think about when if-syntax we allow when we write the new parser.